src/hooks/users.ts

```diff
@@ -1,60 +1,132 @@
+  // }, [isRelatedVenuesLoaded, relatedVenues]);
+
+  // const usersQuery = useMemo<SparkleRFQConfig | undefined>(() => {
+  //   if (!isRelatedVenuesLoaded || !venueId) return;
+
+  //   const universeUserVenueIds = relatedVenues.map((venue) => venue.id);
+
+  //   console.log("venues", relatedVenues);
+
+  //   return {
+  //     collection: "users",
+  //     where: ["enteredVenueIds", "array-contains-any", universeUserVenueIds],
+  //     storeAs: "universeUsers",
+  //   };
+  // }, [venueId, isRelatedVenuesLoaded]);
+
+  // useSparkleFirestoreConnect(usersQuery);
 
-export const useVenueUsers = (): readonly WithId<User>[] => {
-  useConnectVenueUsers();
+  // if (!isRelatedVenuesLoaded || !venueId) return;
 
-  return useSelector(usersSelector) ?? [];
+  const queryFn = (): SparkleRFQConfig[] => {
+    if (!isRelatedVenuesLoaded) return [];
+
+    const universeUserVenueIds = relatedVenues.map((venue) => venue.id);
+
+    return [
+      {
+        collection: "users",
+        where: ["enteredVenueIds", "array-contains", universeUserVenueIds[1]],
+        storeAs: "universeUsers",
+      },
+    ];
+  };
+
+  useSparkleFirestoreConnect(queryFn);
+
+  // useSparkleFirestoreConnect(
+  //   venueId && isRelatedVenuesLoaded
+  //     ? {
+  //         collection: "users",
+  //         where: ["enteredVenueIds", "array-contains", universeUserVenueIds[1]],
+  //         storeAs: "universeUsers",
+  //       }
+  //     : undefined
+  // );
 };
 
-export const usePartygoers = (): readonly WithId<User>[] => {
-  const lastSeenThreshold = useUserLastSeenThreshold();
+export const useUniverseUsers = () => {
+  useConnectUniverseUsers();
 
-  const venueUsers = useVenueUsers();
+  const selectedUniverseUsers = useSelector(universeUsersSelector);
+
+  console.log("users", selectedUniverseUsers);
 
   return useMemo(
-    () => venueUsers.filter((user) => user.lastSeenAt > lastSeenThreshold),
-    [venueUsers, lastSeenThreshold]
+    () => ({
+      universeUsers: selectedUniverseUsers ?? [],
+      isUniverseUsersLoaded: isLoaded(selectedUniverseUsers),
+    }),
+    [selectedUniverseUsers]
   );
 };
 
-export const useIsVenueUsersLoaded = () => {
-  // @debt The reason for this useConnect to be here is that we have entry point components,
-  // which have useConnects calls. And there are checks for the data loaded statuses.
-  // We want to gradualy move from that approach to a more modular one, where the specific data is connected where it is required
-  useConnectVenueUsers();
+// FIXME before merge: will be renamed into useRecentUniverseUsers
+export const usePartygoers = (): readonly WithId<User>[] => {
+  const lastSeenThreshold = useUserLastSeenThreshold();
 
-  const users = useSelector(usersSelector);
+  const { universeUsers } = useUniverseUsers();
 
-  return isLoaded(users);
+  return useMemo(
+    () => universeUsers.filter((user) => user.lastSeenAt > lastSeenThreshold),
+    [universeUsers, lastSeenThreshold]
+  );
```

Comment:
This differs from the logic we used to have, as called out in previous comments elsewhere. Why did we change to using `lastSeenAt` rather than what we had before with `lastSeenIn` and the venue/room/etc? Are we sure that `lastSeenAt` is accurate/updated/correct for it's usage here?

Reply:
Didn't look there so far. Will dive into that after solving `users fetching` issue

Reply:
`lastSeenAt` seems to be a better alternative when speaking about `worldUsers`

Reply:
Fair enough.

> Are we sure that `lastSeenAt` is accurate/updated/correct for it's usage here?

How about this question? Your answer above was ambiguous as to whether we had checked/ensured that it is always updated at the times we need it to be.

Reply:
Good note. I have looked it up and changed its usage in 2 places. However, correct me if I'm wrong, I think we have useInterval updates neither for `lastSeenIn` nor for `lastSeenAt`. We only change them, when person changes the rooms/etc. It doesn't seem right

Reply:
The following locations seem to update `lastSeenAt` / `lastSeenIn`:

- `src/pages/VenuePage/VenuePage.tsx:103`
  - `useInterval(() => { updateUserLocation(); }, LOC_UPDATE_FREQ_MS);`
  - `updateUserLocation` appears to update both `lastSeenIn` and `lastSeenAt`
- `src/utils/useLocationUpdateEffect.ts:37`
  - `updateLocationData` appears to update both `lastSeenIn` and `lastSeenAt`, which is then called by `enterLocation` (`src/utils/useLocationUpdateEffect.ts:51`)
- `src/utils/useLocationUpdateEffect.ts:101`
  - `leaveRoom` calls `updateUserProfile` to update both `lastSeenIn` and `lastSeenAt`
  
Given that `VenuePage` is our main 'entry point' into rendering all of the venues on the 'user' (not admin) side of things, the `useInterval` there should run all the time, so should keep both of these up to date.